client/src/components/ShrinkImage/index.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { Transition } from 'react-transition-group';
+import { Theme, createStyles, WithStyles, withStyles } from '@material-ui/core';
+import classNames from 'classnames';
+
+type ShiftedClassName =
+  | 'entering-shift'
+  | 'entered-shift'
+  | 'exiting-shift'
+  | 'exited-shift'
+  | 'unmounted-shift';
+
+const styles = (theme: Theme) =>
+  createStyles({
+    shrinkimage: {
+      display: 'block',
+      height: '100%',
+      maxWidth: '80vw',
+      maxHeight: '80vh',
+      margin: `${theme.spacing(4)}px auto`,
+      transition: 'transform 300ms ease-in-out',
+    },
+    entering: {
+      transform: 'scale(1.25)',
+    },
+    'entering-shift': {
+      transform: 'translateY(100px) scale(1.25)',
+    },
+    entered: {
+      transform: 'scale(1)',
+    },
+    'entered-shift': {
+      transform: 'translateY(0) scale(1)',
+    },
+    exiting: {},
+    'exiting-shift': {},
+    exited: {},
+    'exited-shift': {},
+    unmounted: {},
+    'unmounted-shift': {},
+  });
+
+interface Props
+  extends WithStyles<typeof styles>,
+    React.ImgHTMLAttributes<HTMLImageElement> {
+  shift?: boolean;
+}
+
+const ShrinkImage: React.FC<Props> = ({
+  classes,
+  shift,
+  ...imgProps
+}: Props) => {
+  return (
+    <Transition in appear timeout={1000}>
+      {(state) => (
+        <img
+          className={classNames(classes.shrinkimage, classes[state], {
+            [classes[`${state}-shift` as ShiftedClassName]]: shift,
```

Comment:
Needed this cast because TS widens `${state}-shift` to string :(

Reply:
all g in the hood